packages/core/core/src/resolveOptions.js

```diff
@@ -146,6 +146,7 @@ export default async function resolveOptions(
   };
 
   let port = determinePort(initialOptions.serveOptions, env.PORT);
+  let hmrPort = determinePort(initialOptions.hmrOptions, env.HMR_PORT);
```

Comment:
Hi, I suggest using `env.HMR_PORT?.length ? env.HMR_PORT : env.PORT`. As such, when only setting `env.PORT`, it would allow for HMR to use the same port when `env.HMR_PORT` isn't set. This is aligned with the behavior of the CLI, where `--port` sets both the main and HMR port.

Reply:
Good idea.

Reply:
Hi @700software , are you interested to implement my suggestion to your PR?